sys/arduino/serialport.cpp

```diff
@@ -249,31 +252,32 @@ size_t SerialPort::println(void)
 
 int SerialPort::read(void)
 {
-    int res = -1;
-
-    irq_disable();
-    if (rx_buf.avail > 0) {
-        res = ringbuffer_get_one(&rx_buf);
+    if (stdio_avail()) {
+        char c;
+        ssize_t n = stdio_read((void*)&c, 1);
```

Comment:
In case of `stdio_uart` backend, this requires to enable module `stdio_uart_rx` to work.